eng/tools/lint-diff/src/runChecks.ts

```diff
@@ -34,63 +35,70 @@ export async function runChecks(
     // If the tags array is empty run the loop once but with a null tag
     const coalescedTags = tags?.length ? tags : [null];
     for (const tag of coalescedTags) {
-      let tagArg = tag ? `--tag=${tag} ` : "";
-
-      let autorestCommand =
-        `npm exec --no -- autorest ` +
-        `--v3 ` +
-        `--spectral ` +
-        `--azure-validator ` +
-        `--semantic-validator=false ` +
-        `--model-validator=false ` +
-        `--message-format=json ` +
-        `--openapi-type=${openApiType} ` +
-        `--openapi-subtype=${openApiSubType} ` +
-        `--use=${dependenciesDir} ` +
-        `${tagArg} ` +
-        `${changedFilePath}`;
-
       console.log(`::group::Autorest for type: ${openApiType} readme: ${readme} tag: ${tag}`);
+
+      const autorestArgs = [
+        "autorest",
+        "--v3",
+        "--spectral",
+        "--azure-validator",
+        "--semantic-validator=false",
+        "--model-validator=false",
+        "--message-format=json",
+        `--openapi-type=${openApiType}`,
+        `--openapi-subtype=${openApiSubType}`,
+        `--use=${dependenciesDir}`,
+      ];
+
+      if (tag) {
+        autorestArgs.push(`--tag=${tag}`);
+      }
+      autorestArgs.push(changedFilePath);
+
+      const autorestCommand = `npm ${autorestArgs.join(" ")}`;
       console.log(`\tAutorest command: ${autorestCommand}`);
 
-      const executionResult = await executeCommand(autorestCommand);
-      
-      console.log(executionResult.stderr + executionResult.stdout);
-
-      const lintDiffResult = {
-        autorestCommand,
-        rootPath: path,
-        readme,
-        tag: tag ? tag : "",
-        openApiType,
-        ...executionResult,
-      };
-      logAutorestExecutionErrors(lintDiffResult);
+      let lintDiffResult: AutorestRunResult;
+      try {
+        const executionResult = await execNpmExec(autorestArgs, { maxBuffer: MAX_EXEC_BUFFER });
```

Comment:
Rather than calling console.log yourself, how about passing an options bag with a ConsoleLogger(), so exec.js can handle the logging for you?  You could set debug=false by default, or opt-in to debug=true if LintDiff exposes debug true/false option.